### ISSUE TITLE
After upgrading to version 4, my datatable no longer renders

### ISSUE THREAD

**Issue by JoshuaVSherman** [gregnb/mui-datatables#1809]
## Expected Behavior
The table should render as it did with version 3.7.8
I have the necessary packages installed and also @mui/styles
Also, I do not find anyplace in documentation that explains what are breaking changes that would need to be done after upgrading.

## Current Behavior
```
getStylesCreator.js:24 MUI: The `styles` argument provided is invalid.
You are providing a function without a theme in the context.
One of the parent elements needs to use a ThemeProvider.
create @ getStylesCreator.js:24
attach @ makeStyles.js:95
(anonymous) @ makeStyles.js:233
useSynchronousEffect @ makeStyles.js:187
useStyles @ makeStyles.js:225
WithStyles @ withStyles.js:56
renderWithHooks @ react-dom.development.js:14985
updateForwardRef @ react-dom.development.js:17044
beginWork @ react-dom.development.js:19098
beginWork$1 @ react-dom.development.js:23940
performUnitOfWork @ react-dom.development.js:22776
workLoopSync @ react-dom.development.js:22707
renderRootSync @ react-dom.development.js:22670
performSyncWorkOnRoot @ react-dom.development.js:22293
scheduleUpdateOnFiber @ react-dom.development.js:21881
updateContainer @ react-dom.development.js:25482
(anonymous) @ react-dom.development.js:26021
unbatchedUpdates @ react-dom.development.js:22431
legacyRenderSubtreeIntoContainer @ react-dom.development.js:26020
render @ react-dom.development.js:26103
./src/index.tsx @ index.tsx:7
__webpack_require__ @ bootstrap:19
(anonymous) @ startup:4
__webpack_require__.O @ chunk loaded:23
(anonymous) @ startup:5
(anonymous) @ startup:5
getStylesCreator.js:24 MUI: The `styles` argument provided is invalid.
You are providing a function without a theme in the context.
One of the parent elements needs to use a ThemeProvider.
create @ getStylesCreator.js:24
attach @ makeStyles.js:95
(anonymous) @ makeStyles.js:233
useSynchronousEffect @ makeStyles.js:187
useStyles @ makeStyles.js:225
WithStyles @ withStyles.js:56
renderWithHooks @ react-dom.development.js:14985
updateForwardRef @ react-dom.development.js:17044
beginWork @ react-dom.development.js:19098
callCallback @ react-dom.development.js:3945
invokeGuardedCallbackDev @ react-dom.development.js:3994
invokeGuardedCallback @ react-dom.development.js:4056
beginWork$1 @ react-dom.development.js:23964
performUnitOfWork @ react-dom.development.js:22776
workLoopSync @ react-dom.development.js:22707
renderRootSync @ react-dom.development.js:22670
performSyncWorkOnRoot @ react-dom.development.js:22293
scheduleUpdateOnFiber @ react-dom.development.js:21881
updateContainer @ react-dom.development.js:25482
(anonymous) @ react-dom.development.js:26021
unbatchedUpdates @ react-dom.development.js:22431
legacyRenderSubtreeIntoContainer @ react-dom.development.js:26020
render @ react-dom.development.js:26103
./src/index.tsx @ index.tsx:7
__webpack_require__ @ bootstrap:19
(anonymous) @ startup:4
__webpack_require__.O @ chunk loaded:23
(anonymous) @ startup:5
(anonymous) @ startup:5
MUIDataTable.js:61 Uncaught TypeError: Cannot read properties of undefined (reading 'down')
    at defaultTableStyles (MUIDataTable.js:61)
    at Object.create (getStylesCreator.js:19)
    at attach (makeStyles.js:95)
    at makeStyles.js:233
    at useSynchronousEffect (makeStyles.js:187)
    at useStyles (makeStyles.js:225)
    at WithStyles (withStyles.js:56)
    at renderWithHooks (react-dom.development.js:14985)
    at updateForwardRef (react-dom.development.js:17044)
    at beginWork (react-dom.development.js:19098)
```

## Steps to Reproduce (for bugs)
I unfortunately cannot provide as it is a private repo, here is some code

```
import React from 'react';
import MUIDataTable, { MUIDataTableColumn } from 'mui-datatables';
import HtmlReactParser from 'html-react-parser';

const makeColumns = (cols: { name: string; label: string; setCellProps: (...args: any) => any }[]): MUIDataTableColumn[] => {
  const columns: MUIDataTableColumn[] = [];
  for (let i = 0; i < cols.length; i += 1) {
    const c = cols[i]; // eslint-disable-line security/detect-object-injection
    columns.push({
      name: c.name,
      label: c.label,
      options: {
        filter: false,
        sort: false,
        setCellProps: (cellValue, rowIndex) => c.setCellProps(cellValue, rowIndex),
        setCellHeaderProps: (value) => ({
          style: { color: 'inherit', fontWeight: 'bold', fontSize: '10pt', fontStyle: 'italic', width: c.name === 'numMethods' ? '150px' : 'auto' },
        }),
        customBodyRender: (value) => {
          if (c.name === 'numMethods') value = `${value}`;
          return (
            <div key={i.toString()} style={{ minWidth: '.65in', margin: 0, fontSize: '12pt', cursor: 'pointer' }}>
              {HtmlReactParser(value as string)}
            </div>
          );
        },
      },
    });
  }
  return columns;
};

export const runRowClick = (row, rowClicks, setSelectedRow) => {
  rowClicks.forEach((r, index) => {
    if (row[0].props.children === r.role.name) {
      r.onClick();
      setSelectedRow({ role: r.role, index });
    }
  });
};

export const checkIfSelectedRow = (row, selectedRow, pageRole) => {
  if (selectedRow.role && selectedRow.role.provenance === pageRole && row[0].props.children === selectedRow.role.name)
    return { style: { backgroundColor: 'rgb(0, 125, 188)', fontWeigt: 'bold', color: 'white' } };
};

export const setFontIfRowWasClicked = (value, rowIndex, selectedRow, pageRole) => {
  if (
    selectedRow.role &&
    pageRole === selectedRow.role.provenance &&
    selectedRow.index === rowIndex &&
    (selectedRow.role.name === value.props.children || `${selectedRow.role.allowedMethods.length}` === value.props.children)
  ) {
    return { style: { color: 'white', fontWeight: 'bold' } };
  }
};

export const RolesTable = ({ rolesTableData, rowClicks, pageRole }): JSX.Element => {
  const [selectedRow, setSelectedRow] = React.useState({});
  return (
    <div id="roles-table">
      <MUIDataTable
        options={{
          filterType: 'dropdown',
          pagination: false,
          responsive: 'standard',
          filter: false,
          download: false,
          search: false,
          print: false,
          viewColumns: false,
          selectableRows: 'none',
          fixedHeader: true,
          setRowProps: (row) => checkIfSelectedRow(row, selectedRow, pageRole),
          onRowClick: (row) => runRowClick(row, rowClicks, setSelectedRow),
        }}
        columns={makeColumns([
          { name: 'name', label: 'Name', setCellProps: (name, rowIndex) => setFontIfRowWasClicked(name, rowIndex, selectedRow, pageRole) },
          {
            name: 'numMethods',
            label: 'Number of Methods',
            setCellProps: (value, rowIndex) => setFontIfRowWasClicked(value, rowIndex, selectedRow, pageRole),
          },
        ])}
        data={rolesTableData}
        title=""
      />
    </div>
  );
};


```

## Your Environment
React 17, node 14

**Comment by JoshuaVSherman** [comment]
I have an open source project and again it fails to render, here is the example, https://github.com/WebJamApps/JaMmusic/blob/717_mui-datatable-upgrade/src/components/TourTable/index.tsx

Also, this could be related to https://github.com/gregnb/mui-datatables/issues/1810

**Comment by AppElent** [comment]
I have the same issue!

**Comment by thefat32** [comment]
You need to add a default theme to your app

```ts
const theme = createTheme();

<ThemeProvider theme={theme}>
    <App />
</ThemeProvider>
```

**Comment by JoshuaVSherman** [comment]
@thefat32 this is not in mui version 5 and from what I found the one they have is not the same for version 5 it is not compatibly with React.strictmode and has been depricated, please let me know when MUI datatables will be fully ready for mui version 5, thank you

**Comment by JoshuaVSherman** [comment]
I think i'm just going to migrate to using https://mui.com/components/data-grid/

i'll go ahead and close this issue now

**Comment by thefat32** [comment]
@JoshuaVSherman your issue was about not being able to run your project after upgrading to v4. The code I shared fixes your problem and is compatible with mui 5. Yes @mui/styles is deprecated. Yes it is not compatible with React's StrictMode, you probably will get warnings when using it but it will not affect production builds at all. It is very common to have this kind of tradeoffs in community projects that depends on the free time the manantainers and collaborators can give to it.

**Comment by zamgill** [comment]
I was able to fix this by installing `@mui/styles@5.0.2`. I am using mui v5.0.3.
